common/djangoapps/util/milestones_helpers.py

```diff
@@ -150,6 +151,31 @@ def fulfill_course_milestone(course_key, user):
             add_user_milestone({'id': user.id}, milestone)
 
 
+def get_required_content(course, user):
+    """
+    Queries milestones subsystem to see if the specified course is gated on one or more milestones,
+    and if those milestones can be fulfilled via completion of a particular course content module
+    """
+    required_content = []
+    if settings.FEATURES.get('MILESTONES_APP', False):
+        # Get all of the outstanding milestones for this course, for this user
+        try:
+            milestone_paths = get_course_milestones_fulfillment_paths(
+                unicode(course.id),
+                serialize_user(user)
+            )
+        except InvalidMilestoneRelationshipTypeException:
+            return required_content
+
+        # For each outstanding milestone, see if this content is one of its fulfillment paths
+        for path_key in milestone_paths:
+            milestone_path = milestone_paths[path_key]
+            if milestone_path.get('content') and len(milestone_path['content']):
```

Comment:
nitpick: Is there any reason why 'content' might not be in this dict? If not, why this check? Also just doing 'and milestone_path['content']' achieves the 2nd part (no need for len to test for empty collection)